app/femr/ui/views/history/indexEncounter.scala.html

```diff
@@ -177,15 +177,32 @@
                             <span class="value">@outputStringOrNA(viewModelPharmacy.getProblems.get(problemIterator - 1))</span></p>
                     }
 
-                    @for(prescription <- viewModelPharmacy.getPrescriptions){
-                        <p>
-                            <span class="infoLabel">Dispensed Medication:</span>
-                            @prescription.getAmount @outputStringOrNA(prescription.getName) (@outputStringOrNA(prescription.getMedicationForm).toString().trim)
-                            @for(activeDrug <- prescription.getMedicationActiveDrugs) {
-                                <span class='activeDrug'> @activeDrug.getName @activeDrug.getValue@activeDrug.getUnit </span>
-                            }
+                    @* AJ Saclayan Prescribed Medications *@
+                    @if(viewModelPharmacy.getOriginalMedications.size != null){
+                       <p> <span class="infoLabel">Originally Prescribed Medication:</span></p>
+                        @for(x <- 1 to viewModelPharmacy.getOriginalMedications.size -1) {
+@*//                           defining(viewModelPharmacy.getOriginalMedications.get(x - 1)) { medication =>*@
 
-                        </p>
+                            <p> <i>@viewModelPharmacy.getOriginalMedications.get(x-1).getName</i></p>
+
+                            <p> Replaced with: </p>
+                            <p>@viewModelPharmacy.getOriginalMedications.get(x-1 +1).getName</p>
```

Comment:
My thought was more along the lines of having a PrescriptionItem carry both the actual dispensed prescription and have it carry the original prescription, if needed. This would mean that wherever you need it you could just call something like "prescriptionItem.getDispensedPrescription()" and if it is a replaced prescription you could also call "prescriptionItem.getOriginalPrescription()" to get the original.

There's a mapper that is used to create all the ____Item objects, you can find the PrescriptionItem creation here:

https://github.com/FEMR/femr/blob/d3539e591d8d75fb2bb9aa363ff5e8ffccb614ff/app/femr/common/ItemModelMapper.java#L253

Of course, if this doesn't help you feel free to completely disregard!